cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/loadbalancer/AzureLoadBalancingRule.java

```diff
@@ -11,11 +14,15 @@
 
     private final AzureLoadBalancerProbe probe;
 
-    public AzureLoadBalancingRule(TargetGroupPortPair portPair) {
+    private final String groupName;
+
+    public AzureLoadBalancingRule(TargetGroupPortPair portPair, Group group) {
         this.backendPort = portPair.getTrafficPort();
         this.frontendPort = portPair.getTrafficPort();
         this.name = defaultNameFromPort(portPair.getTrafficPort());
         this.probe = new AzureLoadBalancerProbe(portPair.getHealthCheckPort());
+        checkNotNull(group);
+        this.groupName = group.getName();
```

Comment:
`this.groupName = checkNotNull(group, "Expected non-null 'checkNotNull(group)'").getName();`